ui/src/components/chat/compose.rs

```diff
@@ -448,166 +481,244 @@ fn get_messages(cx: Scope<ComposeProps>) -> Element {
-                                                },
-                                                on_edit: move |update: String| {
-                                                    edit_msg.set(None);
-                                                    let msg = update.split('\n').collect::<Vec<_>>();
-                                                    let is_valid = msg.iter().any(|x| !x.trim().is_empty());
-                                                    let msg = msg.iter().map(|x| x.to_string()).collect();
-                                                    if !is_valid {
-                                                        ch.send(MessagesCommand::DeleteMessage { conv_id: message.inner.conversation_id(), msg_id: message.inner.id() });
-                                                    }
-                                                    else {
-                                                        ch.send(MessagesCommand::EditMessage { conv_id: message.inner.conversation_id(), msg_id: message.inner.id(), msg})
-                                                    }
-                                                }
-                                            },
-                                       }
-                                    }
-                                )
-                            })
-                        }
-                    )
+                rsx!(render_message_groups{
+                    groups: group_messages(data.my_id.did_key(), *num_to_take.get(),  &data.active_chat.messages),
+                    active_chat_id: data.active_chat.id,
                 })
             }
+        }
+    ))
+}
+
+#[derive(Props)]
+struct AllMessageGroupsProps<'a> {
+    groups: Vec<MessageGroup<'a>>,
+    active_chat_id: Uuid,
+}
+
+// attempting to move the contents of this function into the above rsx! macro causes an error: cannot return vale referencing
+// temporary location
+fn render_message_groups<'a>(cx: Scope<'a, AllMessageGroupsProps<'a>>) -> Element<'a> {
+    log::trace!("render message groups");
+    cx.render(rsx!(cx.props.groups.iter().map(|_group| {
+        rsx!(render_message_group {
+            group: _group,
+            active_chat_id: cx.props.active_chat_id,
+        })
+    })))
+}
+
+#[derive(Props)]
+struct MessageGroupProps<'a> {
+    group: &'a MessageGroup<'a>,
+    active_chat_id: Uuid,
+}
+
+fn render_message_group<'a>(cx: Scope<'a, MessageGroupProps<'a>>) -> Element<'a> {
+    let state = use_shared_state::<State>(cx)?;
+
+    let MessageGroupProps {
+        group,
+        active_chat_id: _,
+    } = cx.props;
+
+    let messages = &group.messages;
+    let last_message = messages.last().unwrap().message;
+    let sender = state.read().get_identity(&group.sender);
+    let sender_name = sender.username();
+    let active_language = &state.read().settings.language;
+
+    // todo: investigate why updating one's profile picture affects their status.
```

Comment:
Is this still the case since #430 was merged?

Reply:
it doesn't look like it. I removed the comment.